# <center>SUY DIỄN QUẦN THỂ</center>
<img src="../images/dang1.png" width=300>

## I. CENTRAL LIMIT THEOREM
Với 1 quần thể bất kỳ - chưa rõ phân phối thế nào, ta chọn hàng loạt mẫu từ quần thể đó (Mẫu A, mẫu B, ...) thì con số trung bình của các mẫu đó ( trung bình A, trung bình B, ..) phân phối theo phân phối bình thường.  
<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/7b/IllustrationCentralTheorem.png/1920px-IllustrationCentralTheorem.png">  

### Các thuộc tính của phân phối bình thường của trung bình:
- Phân phối của quần thể không nhất thiết phải normal
- Mẫu có số mẫu càng lớn thì phân phối càng hẹp lại
- Tương quan với trung bình và độ biến thiên của quần thể

---
## II. Suy diễn số trung bình của quần thể
- Quần thể:
    - Trung bình: M
- Mẫu:
    - Trung bình: $\mu$
    - Độ lệch chuẩn: $\sigma$

Khoảng tin cậy 95% của số trung bình là:  
$$ M = \mu \pm 1.96 \frac{\sigma}{\sqrt{n}}$$
(1.96 thì tương đương PDF của normal distribution = 95%, 2 thì PDF=95.6%)

---
## III. Suy diễn tỉ lệ của quần thể
- Quần thể:
    - Tỉ lệ: PROPORTION
- Mẫu:
    - Tỉ lệ: p

Khoảng tin cậy 95% của tỉ lệ là:  
$$ \text{PROPORTION} = p \pm 1.96 \sqrt{\frac{p(1-p)}{n}}$$

---
## IV. Tuy nhiên, điều kiện để dùng công thức này là cỡ mẫu $\ge$ 30
Nếu cỡ mẫu < 30 thì ta phải thay con số 1.96 bằng kết quả  
Percent point function (ppf) của student-t distribution có degree of freedom là **n-1**. (n là cỡ mẫu)  
**Khi n$\ge$30 thì phân phối student-t = phân phối normal.**

Đối với **proportion** thì điều kiện là np$\geq$5 và n(1-p)$\geq$5.  

|Type      | Condition to use 1.96 - PDF(normal-distribution) =95%)|
|----------|--------------------------------|
|Mean      |$ n\geq 30 $                    |
|Proportion|$ \text{min}[np, n(1-p)]\geq 5 $|

---
## V. Thực hành

In [13]:
import numpy as np
import scipy.stats as st
import matplotlib.pyplot as plt

### A. Khởi tạo standard normal distribution ( trung bình = 0, độ lệch chuẩn = 1)

In [20]:
normal_distribution = st.norm()

##### Tính PDF của phân phối ở khoảng -1.96 -> 1.96 (=95%) bằng cách lấy hiệu của CDF(1.96)- CDF(-1.96)

In [16]:
normal_distribution.cdf(1.96)- normal_distribution.cdf(-1.96)

0.950004209703559

##### Có thể tính ngược lại bằng PPF(0.025)=-1.96 và PPF(0.975)=1.96. (0.975 - 0.025 = 95%)

In [19]:
normal_distribution.ppf(0.025)

-1.9599639845400545

In [21]:
normal_distribution.ppf(0.975)

1.959963984540054

### B. Khởi tạo student-t distribution (độ tự do =9, trung bình =0, độ lệch chuẩn =1)

In [23]:
t_dist = st.t(df=9) # df là degree of freedom = n-1

##### Tính PPF(0.025) và PPF(0.975)

In [24]:
t_dist.ppf(0.025)

-2.262157162740992

In [25]:
t_dist.ppf(0.975)

2.2621571627409915

**=> Suy diễn quần thể sẽ dùng con số 2.26 thay cho 1.96** với cỡ mẫu=10 (df=9), điều này nghĩa là khoảng tin cậy của mình sẽ rộng hơn vì cỡ mẫu nhỏ